README.md
improved version of my crwn repo https://www.github.com/charlz-s/crwn, with more functionality.

changelog.md
<!-- note: this file would be used to document important changes -->

1. commit message: change previous process
changes:
- move the default items from js/defaultItems.js to [json/default.json](json/default.json)
- delete js/defaultItems.js
- added comments to [js/floatingCart.js](js/floatingCart.js)
- moved code from js/cards.js to [js/default.js](js/default.js) and use fetch, async & await to generate the default items from json/default.json
- display the links header in [index.html](index.html) + css tweaks
- delete script.js (would re-write it's functionality)
- create [changelog.md](changelog.md)

-------------

2. commmit message: remove .json
changes:
- remove json/default.json file i cannot send post requests since it's just a file in my local repo
(i should have thought about this before trying to use fetch, but i didn't 🤦🏽‍♂️)
- move the default items data to an array in [js/default.js](js/default.js), and run a for each loop to visualize items
- started work on "add to cart" functionality (still incomplete)


js/cards.js
overlay.setAttribute('class', 'popup-overlay')

defaultItems.forEach(function(item, index){
   let newItem = document.createElement('div')
   newItem.setAttribute('class', 'item-card')

   newItem.innerHTML = `
      <div class="imageDiv">
      </div>
      <div class="textDiv">
         <div>
            <p>${item.name}</p>
            <h3>₦<span>${item.price}</span></h3>
            <div class="description"></div>
         </div>
         <button>Add to Cart</button>
      </div>
      <aside>
         <img class="expand" src="img/expand.svg" alt="expand the item card">
      </aside>
   `

   newItem.firstElementChild.style.background = `url('${item.image}')`;
   newItem.firstElementChild.style.backgroundRepeat = "no-repeat";
   newItem.firstElementChild.style.backgroundSize = 'cover';

   allItems.appendChild(newItem)

   
   if (index > 9) {
      newItem.style.display = "none"
   }

   // event to display all items
   seeAllButton.addEventListener('click', function(){
      newItem.style.display = "initial"
      seeAllButton.style.display = "none"
   });
   
   // expand item card
   newItem.addEventListener('click', function(e){
      
      if (e.target.classList.contains('expand')) {
         let popup = document.createElement('div')
         popup.setAttribute('class', 'popup-card')

         popup.innerHTML = `
            <div class="imageDiv"></div>
            <div class="textDiv">
               <h2>${item.name}</h2>
               <p><span>₦</span>${item.price}</p>

               <div class="desc">
                  <h3>Description</h3>
                  <p class="desc-text">
                     Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas.
                  </p>
               </div>

               <button>Add to Cart</button>
               
               <aside>
                  <img class="remove-popup" src="img/cancel.svg" alt="exit the item tag">
               </aside>
            </div>
         `
         
         popup.firstElementChild.style.background = `url('${item.image}')`;
         popup.firstElementChild.style.backgroundRepeat = "no-repeat";
         popup.firstElementChild.style.backgroundSize = 'cover';
         popup.firstElementChild.style.backgroundPosition = '100%';

         // if (mediaWidth.matches) {
         //    popup.firstElementChild.style.background = '';
         // }

         overlay.appendChild(popup)
         overlay.style.display = 'block'
         
         document.body.style.overflowY = 'hidden'
         // overlay.style.overflowY = 'scroll'

         document.body.appendChild(overlay)

      }
   });

   // remove popup
   document.body.addEventListener('click', function(e){
      if (e.target.classList.contains('remove-popup')){
         document.body.style.overflowY = 'scroll';
         overlay.style.display = 'none'
         overlay.innerHTML = "";
      }
   })
})



js/script.js
// // JUNK CODE

class InCartItem {
   constructor (title, price, description, background) {
      this.title = title;
      this.price = price;
      this.desc = description;
      this.background = background;
   }
}

class Cart {
   addNewCartItem(item){
      const cartItem = document.createElement('div')
      cartItem.className = 'cart-item'
      cartItem.innerHTML = `
         <img src="${item.background}">
         <div>
            <p>${item.title}</p>
            <p><span class="item-count">1</span> x ₦${item.price}</p>
         </div>
      `

      const cartList = document.querySelector('#cart-list')
      cartList.appendChild(cartItem)
   }
}


for (let i = 0; i < allItems.children.length; i++){
   allItems.children[i].children[1].children[1].addEventListener('click', (e) => {

      const title = allItems.children[i].children[1].children[0].children[0].textContent;
      const price = Number(allItems.children[i].children[1].children[0].children[1].children[0].textContent);
      const description = allItems.children[i].children[1].children[0].children[2].textContent;
      const background = allItems.children[i].children[0].style.backgroundImage.split('"')[1]

      const newInCart = new InCartItem(title, price, description, background)
      
      const newCartItem = new Cart(newInCart)
      newCartItem.addNewCartItem(newInCart)
      
      document.querySelector('#cart-list').addEventListener('click', (e) => {
         const allCartItem = document.querySelector('#cart-list').children[i].children[0].src
      })
   })

}



js/defaultItems.js
// floating cart icon
let cartIcon = document.querySelector('#cart-icon')
let cartSvg = document.querySelector('.svg')
let cartDiv = document.createElement('div')

// recently added section
let allItems = document.querySelector('#allItems')
let seeAllButton = document.querySelector('#see-all')

// item card overlay
let overlay = document.createElement('div')

// existing objects
let defaultItems = [
   {
      id: 1,
      name: "Brown Brim",
      image: "https://i.ibb.co/ZYW3VTp/brown-brim.png",		
      price: 2700,
      inCart:0
   },
   {
      id: 2,
      name: "Blue Beanie",
      image: "https://i.ibb.co/ypkgK0X/blue-beanie.png",			
      price: 2800,
      inCart:0
   },
   {
      id: 3,
      name: "Brown Cowboy",
      image: "https://i.ibb.co/QdJwgmp/brown-cowboy.png",				
      price: 3500,
      inCart:0
   },
   {
      id: 4,
      name: "Grey Brim",
      image:"https://i.ibb.co/RjBLWxB/grey-brim.png",	
      price: 3000,
      inCart:0
   },
   {
      id: 5,
      name: "Adidas NMD",
      image: "https://i.ibb.co/0s3pdnc/adidas-nmd.png",		
      price: 15000,
      inCart:0
   },
   {
      id: 6,
      name: "Adidas Yeezy",
      image:"https://i.ibb.co/dJbG1cT/yeezy.png",
      price: 18000,
      inCart:0
   },
   {
      id: 7,
      name: "Black Converse",
      image:"https://i.ibb.co/bPmVXyP/black-converse.png",
      price: 13500,
      inCart:0
   },
   {
      id: 8,
      name: "Nike White AirForce",
      image:"https://i.ibb.co/1RcFPk0/white-nike-high-tops.png",
      price: 16000,
      inCart:0
   },
   {
      id: 9,
      name: "Black Jean Shearling",
      image:"https://i.ibb.co/XzcwL5s/black-shearling.png",
      price: 12500,
      inCart:0
   },
   {
      id: 10,
      name: "Blue Jean Jacket",
      image:"https://i.ibb.co/mJS6vz0/blue-jean-jacket.png",
      price: 9000,
      inCart:0
   },
   {
      id: 11,
      name: "Grey Jean Jacket",
      image:"https://i.ibb.co/N71k1ML/grey-jean-jacket.png",
      price: 9000,
      inCart:0
   },
   {
      id: 12,
      name: "Brown Shearling",
      image:"https://i.ibb.co/s96FpdP/brown-shearling.png",
      price: 9500,
      inCart:0
   },
   {
      id: 13,
      name: "Blue Tanktop",
      image:"https://i.ibb.co/7CQVJNm/blue-tank.png",	
      price: 3800,
      inCart:0
   },
   {
      id: 14,
      name: "Floral Blouse",
      image:"https://i.ibb.co/4W2DGKm/floral-blouse.png",
      price: 4500,
      inCart:0
   },
   {
      id: 15,
      name: "Floral Dress",
      image:"https://i.ibb.co/KV18Ysr/floral-skirt.png",				
      price: 4500,
      inCart:0
   },
   {
      id: 16,
      name: "Red Dots Dress",
      image:"https://i.ibb.co/N3BN1bh/red-polka-dot-dress.png",
      price: 4000,
      inCart:0
   },
   {
      id: 17,
      name: "Camo Down Vest",
      image:"https://i.ibb.co/xJS0T3Y/camo-vest.png",	
      price: 7250,
      inCart:0
   },
   {
      id: 18,
      name: "Floral T-shirt",
      image:"https://i.ibb.co/qMQ75QZ/floral-shirt.png",				
      price: 3750,
      inCart:0
   },
   {
      id: 19,
      name: "Black & White Longsleeve",
      image:"https://i.ibb.co/55z32tw/long-sleeve.png",			
      price: 5500,
      inCart:0
   },
   {
      id: 20,
      name: "Pink T-shirt",
      image:"https://i.ibb.co/RvwnBL8/pink-shirt.png",		
      price: 4500,
      inCart:0
   }
]

js/default.js
// recently added section
let allItems = document.querySelector('#allItems')
let seeAllButton = document.querySelector('#see-all')

 // item card overlay
 let overlay = document.createElement('div')

 overlay.setAttribute('class', 'popup-overlay')


const defaultItems = [
   {
      "id": 1,
      "name": "Brown Brim",
      "image": "https://i.ibb.co/ZYW3VTp/brown-brim.png",		
      "price": 2700,
      "inCart": 0,
      "catergory": "hat",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 2,
      "name": "Blue Beanie",
      "image": "https://i.ibb.co/ypkgK0X/blue-beanie.png",			
      "price": 2800,
      "inCart": 0,
      "catergory": "hat",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 3,
      "name": "Brown Cowboy",
      "image": "https://i.ibb.co/QdJwgmp/brown-cowboy.png",				
      "price": 3500,
      "inCart": 0,
      "catergory": "hat",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 4,
      "name": "Grey Brim",
      "image":"https://i.ibb.co/RjBLWxB/grey-brim.png",	
      "price": 3000,
      "inCart": 0,
      "catergory": "hat",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 5,
      "name": "Adidas NMD",
      "image": "https://i.ibb.co/0s3pdnc/adidas-nmd.png",		
      "price": 15000,
      "inCart": 0,
      "catergory": "sneakers",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 6,
      "name": "Adidas Yeezy",
      "image":"https://i.ibb.co/dJbG1cT/yeezy.png",
      "price": 18000,
      "inCart": 0,
      "catergory": "sneakers",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 7,
      "name": "Black Converse",
      "image":"https://i.ibb.co/bPmVXyP/black-converse.png",
      "price": 13500,
      "inCart": 0,
      "catergory": "sneakers",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 8,
      "name": "Nike White AirForce",
      "image":"https://i.ibb.co/1RcFPk0/white-nike-high-tops.png",
      "price": 16000,
      "inCart": 0,
      "catergory": "sneakers",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 9,
      "name": "Black Jean Shearling",
      "image":"https://i.ibb.co/XzcwL5s/black-shearling.png",
      "price": 12500,
      "inCart": 0,
      "catergory": "jacket",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 10,
      "name": "Blue Jean Jacket",
      "image":"https://i.ibb.co/mJS6vz0/blue-jean-jacket.png",
      "price": 9000,
      "inCart": 0,
      "catergory": "jacket",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 11,
      "name": "Grey Jean Jacket",
      "image":"https://i.ibb.co/N71k1ML/grey-jean-jacket.png",
      "price": 9000,
      "inCart": 0,
      "catergory": "jacket",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 12,
      "name": "Brown Shearling",
      "image":"https://i.ibb.co/s96FpdP/brown-shearling.png",
      "price": 9500,
      "inCart": 0,
      "catergory": "jacket",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 13,
      "name": "Blue Tanktop",
      "image":"https://i.ibb.co/7CQVJNm/blue-tank.png",	
      "price": 3800,
      "inCart": 0,
      "catergory": "women",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 14,
      "name": "Floral Blouse",
      "image":"https://i.ibb.co/4W2DGKm/floral-blouse.png",
      "price": 4500,
      "inCart": 0,
      "catergory": "women",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 15,
      "name": "Floral Dress",
      "image":"https://i.ibb.co/KV18Ysr/floral-skirt.png",				
      "price": 4500,
      "inCart": 0,
      "catergory": "women",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 16,
      "name": "Red Dots Dress",
      "image":"https://i.ibb.co/N3BN1bh/red-polka-dot-dress.png",
      "price": 4000,
      "inCart": 0,
      "catergory": "women",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 17,
      "name": "Camo Down Vest",
      "image":"https://i.ibb.co/xJS0T3Y/camo-vest.png",	
      "price": 7250,
      "inCart": 0,
      "catergory": "men",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 18,
      "name": "Floral T-shirt",
      "image":"https://i.ibb.co/qMQ75QZ/floral-shirt.png",				
      "price": 3750,
      "inCart": 0,
      "catergory": "men",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 19,
      "name": "Black & White Longsleeve",
      "image":"https://i.ibb.co/55z32tw/long-sleeve.png",			
      "price": 5500,
      "inCart": 0,
      "catergory": "men",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   },
   {
      "id": 20,
      "name": "Pink T-shirt",
      "image":"https://i.ibb.co/RvwnBL8/pink-shirt.png",		
      "price": 4500,
      "inCart": 0,
      "catergory": "men",
      "description": "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Repellat, consectetur? Maxime minima iste reprehenderit saepe nisi doloremque omnis porro officia totam ducimus delectus aspernatur quas."
   }
]

defaultItems.forEach((newItem, index) => {
   let item = document.createElement('div')
   item.setAttribute('class', 'item-card')

   item.innerHTML = `
      <div class="imageDiv">
      </div>
      <div class="textDiv">
         <div>
            <p>${newItem.name}</p>
            <h3>₦<span>${newItem.price}</span></h3>
            <div class="description"></div>
         </div>
         <button class="cart-button">Add to Cart</button>
      </div>
      <aside>
         <img class="expand" src="img/expand.svg" alt="expand the item card">
      </aside>
   `

   
   item.firstElementChild.style.background = `url('${newItem.image}')`;
   item.firstElementChild.style.backgroundRepeat = "no-repeat";
   item.firstElementChild.style.backgroundSize = 'cover';

   allItems.appendChild(item)

   // display only the 4 first items by default
   if (index > 3) {
      item.style.display = "none"
   }

   
   // display all items
   seeAllButton.addEventListener('click', function(){
      item.style.display = "initial"
      seeAllButton.style.display = "none"
   });

   

   // expand item card
   item.addEventListener('click', function(e){
      
      if (e.target.classList.contains('expand')) {
         let popup = document.createElement('div')
         popup.setAttribute('class', 'popup-card')

         popup.innerHTML = `
            <div class="imageDiv"></div>
            <div class="textDiv">
            <h2>${newItem.name}</h2>
            <p><span>₦</span>${newItem.price}</p>

               <div class="desc">
                  <h3>Description</h3>
                  <p class="desc-text">${newItem.description}</p>
               </div>

               <button>Add to Cart</button>
               
               <aside>
                  <img class="remove-popup" src="img/cancel.svg" alt="exit the item tag">
               </aside>
            </div>
         `
         
         popup.firstElementChild.style.background = `url('${newItem.image}')`;
         popup.firstElementChild.style.backgroundRepeat = "no-repeat";
         popup.firstElementChild.style.backgroundSize = 'cover';
         popup.firstElementChild.style.backgroundPosition = '100%';

         // if (mediaWidth.matches) {
         //    popup.firstElementChild.style.background = '';
         // }

         overlay.appendChild(popup)
         overlay.style.display = 'block'
         
         document.body.style.overflowY = 'hidden'
         // overlay.style.overflowY = 'scroll'

         document.body.appendChild(overlay)

            // remove popup
         document.body.addEventListener('click', function(e){
            if (e.target.classList.contains('remove-popup')){
               document.body.style.overflowY = 'scroll';
               overlay.style.display = 'none'
               overlay.innerHTML = "";
            }
         })

      }
   });
         
})